registry/zookeeper/registry.go

```diff
@@ -255,3 +269,39 @@ func (r *zkRegistry) getListener(conf *common.URL) (*RegistryConfigurationListen
 
 	return zkListener, nil
 }
+
+func (r *zkRegistry) getCloseListener(conf *common.URL) (*RegistryConfigurationListener, error) {
+
+	var zkListener *RegistryConfigurationListener
+	r.dataListener.mutex.Lock()
+	configurationListener := r.dataListener.subscribed[conf]
+	if configurationListener != nil {
+
+		zkListener, _ := configurationListener.(*RegistryConfigurationListener)
+		if zkListener != nil {
+			r.listenerLock.Lock()
```

Comment:
It looks like there is not necessary to add lock here because you do not change any system status inside the if-block. It's not the check-do something scenario.

Reply:
this block reference `getListener` , then i not sure why add lock here at this moment. just  stay the same with it.